pyaedt/edb_grpc/core/edb_data/layer_data.py

```diff
@@ -0,0 +1,1753 @@
+        lcNew = self._edb.Cell.LayerCollection()
+        newLayers.Reverse()
+        if not lcNew.AddLayers(newLayers) or not self._pedbstackup._active_layout.SetLayerCollection(lcNew):
+            self._logger.error("Failed to set new layers when updating the stackup information.")
+            return False
+        self._update_edb_objects()
+        return True
+
+
+class LayerEdbClass(object):
+    """Manages Edb Layers. Replaces EDBLayer."""
+
+    def __init__(self, pclass, name):
+        self._pclass = pclass
+        self._name = name
+        self._color = ()
+        self._type = ""
+        self._material = ""
+        self._conductivity = 0.0
+        self._permittivity = 0.0
+        self._loss_tangent = 0.0
+        self._dielectric_fill = ""
+        self._thickness = 0.0
+        self._etch_factor = 0.0
+        self._roughness_enabled = False
+        self._top_hallhuray_nodule_radius = 0.5e-6
+        self._top_hallhuray_surface_ratio = 2.9
+        self._bottom_hallhuray_nodule_radius = 0.5e-6
+        self._bottom_hallhuray_surface_ratio = 2.9
+        self._side_hallhuray_nodule_radius = 0.5e-6
+        self._side_hallhuray_surface_ratio = 2.9
+        self._material = None
+        self._upper_elevation = 0.0
+        self._lower_elevation = 0.0
+
+    @property
+    def lower_elevation(self):
+        """Lower elevation.
+
+        Returns
+        -------
+        float
+            Lower elevation.
+        """
+        try:
+            self._lower_elevation = self._edb_layer.GetLowerElevation()
+        except:
+            pass
+        return self._lower_elevation
+
+    @lower_elevation.setter
+    def lower_elevation(self, value):
+        layer_clone = self._edb_layer
+        layer_clone.SetLowerElevation(self._pclass._edb_value(value))
+        self._pclass._set_layout_stackup(layer_clone, "change_attribute")
+
+    @property
+    def upper_elevation(self):
+        """Upper elevation.
+
+        Returns
+        -------
+        float
+            Upper elevation.
+        """
+        try:
+            self._upper_elevation = self._edb_layer.GetUpperElevation()
+        except:
+            pass
+        return self._upper_elevation
+
+    @property
+    def _edb(self):
+        return self._pclass._pedb.edb
+
+    @property
+    def _edb_layer(self):
+        for l in self._pclass._edb_layer_list:
+            if l.GetName() == self._name:
+                return l.Clone()
+
+    @property
+    def is_stackup_layer(self):
+        """Determine whether this layer is a stackup layer.
+
+        Returns
+        -------
+        bool
+            True if this layer is a stackup layer, False otherwise.
+        """
+        return self._edb_layer.IsStackupLayer()
+
+    @property
+    def is_negative(self):
+        """Determine whether this layer is a negative layer.
+
+        Returns
+        -------
+        bool
+            True if this layer is a negative layer, False otherwise.
```

Comment:
```suggestion
        """Whether this layer is a stackup layer.

        Returns
        -------
        bool
            ``True`` if this layer is a stackup layer, ``False`` otherwise.
        """
        return self._edb_layer.IsStackupLayer()

    @property
    def is_negative(self):
        """Whether this layer is a negative layer.

        Returns
        -------
        bool
            ``True`` if this layer is a negative layer, ``False`` otherwise.
```